### PR TITLE
Add ldap tests

### PR DESCRIPTION
**Pull request by tatankat**
Pull Request for missing LDAP tests as mentioned by @HLeithner in #37962 .

### Summary of Changes

* add ldap service to drone and appveyor
* add certificates
* added tests for authentication against ldap database

### Testing Instructions

run the unit tests with #38421 applied (or an equally patched phpunit.xml)

### Actual result BEFORE applying this Pull Request

no ldap tests (Tests: 784, Assertions: 1264, Skipped: 4.)

### Expected result AFTER applying this Pull Request

5 more tests, 3 more assertions, 2 more skipped test (waiting for PRs #37962 and #37959)

### Documentation Changes Required

Maybe: need to run the ldap docker image to run tests
tests/Unit/README.md should be updated on how to run the unit tests locally. I can add this once the general idea of this PR is approved.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -5,6 +5,37 @@ name: default
...
+        LDAP_USERS=customuser
+        LDAP_PASSWORDS=custompassword
+        LDAP_ENABLE_TLS=yes
+        LDAP_TLS_CERT_FILE=/certificates/ldapcert.crt
+        LDAP_TLS_KEY_FILE=/certificates/ldapcert.key
+        LDAP_TLS_CA_FILE=/certificates/CA.crt
+        BITNAMI_DEBUG=true
+        LDAP_CONFIG_ADMIN_ENABLED=yes
+        LDAP_CONFIG_ADMIN_USERNAME=admin
+        LDAP_CONFIG_ADMIN_PASSWORD=configpassword
```
Comment (HLeithner) [inline .drone.yml]:
```suggestion
      - LDAP_ADMIN_USERNAME=admin
      - LDAP_ADMIN_PASSWORD=adminpassword
      - LDAP_USERS=customuser
      - LDAP_PASSWORDS=custompassword
      - LDAP_ENABLE_TLS=yes
      - LDAP_TLS_CERT_FILE=/certificates/ldapcert.crt
      - LDAP_TLS_KEY_FILE=/certificates/ldapcert.key
      - LDAP_TLS_CA_FILE=/certificates/CA.crt
      - BITNAMI_DEBUG=true
      - LDAP_CONFIG_ADMIN_ENABLED=yes
      - LDAP_CONFIG_ADMIN_USERNAME=admin
      - LDAP_CONFIG_ADMIN_PASSWORD=configpassword
```

File: .drone.yml [hunk 2]
```diff
@@ -5,6 +5,37 @@ name: default
...
+      - LDAP_USERS=customuser
+      - LDAP_PASSWORDS=custompassword
+      - LDAP_ENABLE_TLS=yes
+      - LDAP_TLS_CERT_FILE=/certificates/ldapcert.crt
+      - LDAP_TLS_KEY_FILE=/certificates/ldapcert.key
+      - LDAP_TLS_CA_FILE=/certificates/CA.crt
+      - BITNAMI_DEBUG=true
+      - LDAP_CONFIG_ADMIN_ENABLED=yes
+      - LDAP_CONFIG_ADMIN_USERNAME=admin
+      - LDAP_CONFIG_ADMIN_PASSWORD=configpassword
```
Comment (HLeithner) [inline .drone.yml]:
```suggestion
      - LDAP_ADMIN_USERNAME: admin
      - LDAP_ADMIN_PASSWORD: adminpassword
      - LDAP_USERS: customuser
      - LDAP_PASSWORDS: custompassword
      - LDAP_ENABLE_TLS: yes
      - LDAP_TLS_CERT_FILE: /certificates/ldapcert.crt
      - LDAP_TLS_KEY_FILE: /certificates/ldapcert.key
      - LDAP_TLS_CA_FILE: /certificates/CA.crt
      - BITNAMI_DEBUG: true
      - LDAP_CONFIG_ADMIN_ENABLED: yes
      - LDAP_CONFIG_ADMIN_USERNAME: admin
      - LDAP_CONFIG_ADMIN_PASSWORD: configpassword
```

File: .drone.yml [hunk 3]
```diff
@@ -5,6 +5,37 @@ name: default
...
+      - LDAP_USERS: customuser
+      - LDAP_PASSWORDS: custompassword
+      - LDAP_ENABLE_TLS: yes
+      - LDAP_TLS_CERT_FILE: /certificates/ldapcert.crt
+      - LDAP_TLS_KEY_FILE: /certificates/ldapcert.key
+      - LDAP_TLS_CA_FILE: /certificates/CA.crt
+      - BITNAMI_DEBUG: true
+      - LDAP_CONFIG_ADMIN_ENABLED: yes
+      - LDAP_CONFIG_ADMIN_USERNAME: admin
+      - LDAP_CONFIG_ADMIN_PASSWORD: configpassword
```
Comment (HLeithner) [inline .drone.yml]:
```suggestion
      LDAP_ADMIN_USERNAME: admin
      LDAP_ADMIN_PASSWORD: adminpassword
      LDAP_USERS: customuser
      LDAP_PASSWORDS: custompassword
      LDAP_ENABLE_TLS: yes
      LDAP_TLS_CERT_FILE: /certificates/ldapcert.crt
      LDAP_TLS_KEY_FILE: /certificates/ldapcert.key
      LDAP_TLS_CA_FILE: /certificates/CA.crt
      BITNAMI_DEBUG: true
      LDAP_CONFIG_ADMIN_ENABLED: yes
      LDAP_CONFIG_ADMIN_USERNAME: admin
      LDAP_CONFIG_ADMIN_PASSWORD: configpassword
```

**Comment by tatankat** [comment]
Now it also works on appveyor, thanks to the bitnami team, who accepted my PRs for their openldap image to include more logging so I could find the problem and to fix running as windows Administrator.
It also worked in drone last time it ran there (see https://ci.joomla.org/joomla/joomla-cms/57306).
So, can someone (@HLeithner) (re)take a look at this?

**Comment by HLeithner** [comment]
@Hackwar can you have a look and merge it afterwards? I update the drone signature already

**Pull request merged**